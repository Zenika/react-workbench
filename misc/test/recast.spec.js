/* eslint-env jest */
const { parse } = require('./recast')

jest.mock('recast', () => ({
  parse: jest.fn(source => ({ program: source })),
  types: {
    NodePath: class {
      constructor(param) {
        this.param = param
      }
    },
  },
}))

describe('tests/recast', () => {
  describe('parse', () => {
    it('should take one string containing javascript code', () => {
      const result = parse('const React = require("React")')
      expect(result).toMatchSnapshot()
    })
    it('should take an array string containing javascript code and join it', () => {
      const result = parse(['const React = require("React")', 'const React = require("React")'])
      expect(result).toMatchSnapshot()
    })
  })
})
