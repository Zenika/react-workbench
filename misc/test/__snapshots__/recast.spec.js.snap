// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests/recast parse should take an array string containing javascript code and join it 1`] = `
NodePath {
  "param": "const React = require(\\"React\\")
const React = require(\\"React\\")",
}
`;

exports[`tests/recast parse should take one string containing javascript code 1`] = `
NodePath {
  "param": "const React = require(\\"React\\")",
}
`;
